invenio_rdm_records/services/communities/service.py

```diff
@@ -6,24 +6,100 @@
 # it under the terms of the MIT License; see LICENSE file for more details.
 
 """RDM Record Communities Service."""
+from flask import current_app
+from invenio_records_resources.services import Service, ServiceSchemaWrapper
+from invenio_records_resources.services.uow import RecordCommitOp, unit_of_work
+from kombu import Queue
+from werkzeug.local import LocalProxy
 
-from flask_babelex import lazy_gettext as _
-from invenio_drafts_resources.services.records import RecordService
-from invenio_records_resources.services.uow import unit_of_work
+from invenio_rdm_records.services.errors import (
+    ExceedMaxNumberOfCommunities,
+    RecordCommunityMissing,
+)
 
 
-class RecordCommunitiesService(RecordService):
+class RecordCommunitiesService(Service):
     """Record communities service.
 
     The communities service is in charge of managing communities of a given record.
     """
 
+    @property
+    def indexer(self):
+        """Factory for creating an indexer instance."""
+        return self.config.indexer_cls(
+            # the routing key is mandatory in the indexer constructor since
+            # it is afterwards passed explicitly to the created consumers
+            # and producers. this means that it is not strictly necessary to
+            # pass it to the queue constructor. however, it is passed for
+            # consistency (in case the queue is used by itself) and to help
+            # entity declaration on publish.
+            queue=LocalProxy(
+                lambda: Queue(
+                    self.config.indexer_queue_name,
+                    exchange=current_app.config["INDEXER_MQ_EXCHANGE"],
+                    routing_key=self.config.indexer_queue_name,
+                )
+            ),
+            routing_key=self.config.indexer_queue_name,
+            record_cls=self.config.record_cls,
+            record_to_index=self.record_to_index,
+            record_dumper=self.config.index_dumper,
+        )
+
+    def record_to_index(self, record):
+        """Function used to map a record to an index."""
+        return record.index._name
+
+    @property
+    def schema(self):
+        """Returns the data schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.schema)
+
+    @property
+    def record_cls(self):
+        """Factory for creating a record class."""
+        return self.config.record_cls
+
     @unit_of_work()
     def create(self, identity, data, record, uow=None):
         """Include the record in the given communities."""
         pass
 
+    def _remove(self, community_id, record):
+        """Remove a community from the record."""
+        if community_id not in record.parent.communities.ids:
+            raise RecordCommunityMissing(record.id, community_id)
+
+        record.parent.communities.remove(community_id)
```

Comment:
I forgot to add a comment here explaining that the remove does already remove the default as well. If there are no other comments to address in this PR I will add it in the next PR I am creating.